src/mrinufft/extras/smaps.py

```diff
@@ -194,18 +238,181 @@ def low_frequency(
+        By default it is 0.1
+    backend : str
+        The backend used for the operator for `pinv` computation
+    density : numpy.ndarray, optional
+        The density compensation weights.
+    calib_width : int or tuple of int, optional
+        The calibration region width. By default it is 24.
+    kernel_width : int or tuple of int, optional
+        The kernel width. By default it is 6.
+    thresh : float, optional
+        The threshold for the singular values. By default it is 0.02.
+    crop : float, optional
+        The cropping threshold for the sensitivity maps.
+        By default it is 0.95.
+    decim : int, optional
+        The decimation factor for the caluclation of sensitivity maps.
+        By default it is 1. This can be used to speed up the computation
+        and significantly reduce memory usage. The final result is
+        upsampled back to the original size through linear
+        interpolation.
+
+    Returns
+    -------
+    Smaps : numpy.ndarray
+        The sensitivity maps.
+    """
+    # defer import to later to prevent circular import
+    from mrinufft import get_operator
+    from mrinufft.operators.base import power_method
+
+    try:
+        from skimage.restoration import unwrap_phase
+    except ImportError as err:
+        raise ImportError(
+            "The scikit-image module is not available. Please install "
+            "it along with the [extra] dependencies "
+            "or using `pip install scikit-image`."
+        ) from err
+    if isinstance(calib_width, int):
+        calib_width = (calib_width,) * traj.shape[-1]
+    if isinstance(kernel_width, int):
+        kernel_width = (kernel_width,) * traj.shape[-1]
+    xp = get_array_module(kspace_data)
+    k_space, samples, dc = _extract_kspace_center(
+        kspace_data=kspace_data,
+        kspace_loc=traj,
+        threshold=tuple(float(sh) for sh in calib_width / np.asarray(shape)),
+        density=density,
+        window_fun="rect",
+    )
+    n_coils = k_space.shape[0]
+    central_kspace_img = get_operator(backend)(
+        samples,
+        shape,
+        density=dc,
+        n_coils=k_space.shape[-2],
+        squeeze_dims=True,
+    ).pinv_solver(k_space)
+    central_kspace = fft(central_kspace_img)
+    if decim > 1:
+        central_kspace = _crop_or_pad(
+            central_kspace,
+            tuple(
+                sh // decim if i != 0 else sh
+                for i, sh in enumerate(central_kspace.shape)
+            ),
+        )
+    calib_shape = (n_coils, *calib_width)
+    calib = _crop_or_pad(central_kspace, calib_shape)
+    calib = _unfold_blocks(calib, kernel_width)
+    calib = calib.reshape(
+        calib.shape[0],
+        -1,
+        np.prod(kernel_width),
+    )
+    calib = calib.transpose(1, 0, 2).reshape(-1, calib.shape[0] * np.prod(kernel_width))
+    _, S, VH = xp.linalg.svd(calib, full_matrices=False)
+    VH = VH[S > thresh * S.max(), :]
```

Comment:
If this is a bottleneck, something along these lines could help: 
```python
    max_rank = np.min(calib.shape) -1
    k = np.min(max_rank, 10)
    svds = sp.linalg.sparse.svds if xp == np  else cupyx.scipy.sparse.linalg.svds
    while k < max_rank:
       _, S, VH = svds(calib,k=k full_matrices=False)
       if min(S) < max(S)*thresh: 
           break
       else: 
           k = np.min(max_rank, k+5)
    VH = VH[S > thresh * S.max(), :]
```

Reply:
Is this better? Especially as we need all the eigen vectors in the end anyway

Reply:
it might if you have a very low rank matrix